cmd/mavenExecute.go

```diff
@@ -4,29 +4,40 @@ import (
 	"github.com/SAP/jenkins-library/pkg/command"
 	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/maven"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"io/ioutil"
 
 	"github.com/SAP/jenkins-library/pkg/telemetry"
 )
 
-func mavenExecute(config mavenExecuteOptions, telemetryData *telemetry.CustomData) string {
-	c := command.Command{}
+func mavenExecute(config mavenExecuteOptions, _ *telemetry.CustomData) {
+	runner := command.Command{}
+	err := runMavenExecute(config, &runner)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
 
+func runMavenExecute(config mavenExecuteOptions, runner execRunner) error {
 	options := maven.ExecuteOptions{
 		PomPath:                     config.PomPath,
 		ProjectSettingsFile:         config.ProjectSettingsFile,
 		GlobalSettingsFile:          config.GlobalSettingsFile,
 		M2Path:                      config.M2Path,
-		Goals:                       config.Goals,
-		Defines:                     config.Defines,
-		Flags:                       config.Flags,
+		Goals:                       splitAndTrimParams(config.Goals),
```

Comment:
What is the behavior if the `separator` is not given?

Reply:
There is an indirection via a small private function. This one passes `" "` as separator to the utils function.